security/pkg/nodeagent/cache/secretcache_test.go

```diff
@@ -641,6 +641,34 @@ func TestProxyConfigAnchors(t *testing.T) {
 	})
 }
 
+func TestProxyConfigAnchorsTriggerWorkloadCertUpdate(t *testing.T) {
+	cacheLog.SetOutputLevel(log.DebugLevel)
+	fakeCACli, err := mock.NewMockCAClient(time.Millisecond*1000, false)
+	if err != nil {
+		t.Fatalf("Error creating Mock CA client: %v", err)
+	}
+	u := NewUpdateTracker(t)
+	sc := createCache(t, fakeCACli, u.Callback, security.Options{WorkloadRSAKeySize: 2048})
+	_, err = sc.GenerateSecret(security.WorkloadKeyCertResourceName)
+	if err != nil {
+		t.Errorf("failed to generate certificate for trustAnchor test case")
+	}
+	// Ensure Root cert call back gets invoked once
+	u.Expect(map[string]int{security.RootCertReqResourceName: 1})
```

Comment:
```suggestion
u.Expect(map[string]int{security.RootCertReqResourceName: 1, security.WorkloadKeyCertResourceName: 1})
```

otherwise it is flaky and not really asserting the right thing; we expect to get the workload cert too?

Reply:
updated the test